builtin/providers/aws/resource_aws_db_option_group.go

```diff
@@ -0,0 +1,295 @@
+			},
+			"engine_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"major_engine_version": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"option_group_description": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"option": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"option_name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"port": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+						},
+						"db_security_group_memberships": &schema.Schema{
+							Type:     schema.TypeSet,
+							Optional: true,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+						},
+						"vpc_security_group_memberships": &schema.Schema{
+							Type:     schema.TypeSet,
+							Optional: true,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+						},
+					},
+				},
+				Set: resourceAwsDbOptionHash,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsDbOptionGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	rdsconn := meta.(*AWSClient).rdsconn
+	tags := tagsFromMapRDS(d.Get("tags").(map[string]interface{}))
+
+	createOpts := &rds.CreateOptionGroupInput{
+		EngineName:             aws.String(d.Get("engine_name").(string)),
+		MajorEngineVersion:     aws.String(d.Get("major_engine_version").(string)),
+		OptionGroupDescription: aws.String(d.Get("option_group_description").(string)),
+		OptionGroupName:        aws.String(d.Get("name").(string)),
+		Tags:                   tags,
+	}
+
+	log.Printf("[DEBUG] Create DB Option Group: %#v", createOpts)
+	_, err := rdsconn.CreateOptionGroup(createOpts)
+	if err != nil {
+		return fmt.Errorf("Error creating DB Option Group: %s", err)
+	}
+
+	d.SetId(d.Get("name").(string))
+	log.Printf("[INFO] DB Option Group ID: %s", d.Id())
+
+	return resourceAwsDbOptionGroupUpdate(d, meta)
+}
+
+func resourceAwsDbOptionGroupRead(d *schema.ResourceData, meta interface{}) error {
+	rdsconn := meta.(*AWSClient).rdsconn
+	params := &rds.DescribeOptionGroupsInput{
+		OptionGroupName: aws.String(d.Get("name").(string)),
+	}
+
+	log.Printf("[DEBUG] Describe DB Option Group: %#v", params)
+	options, err := rdsconn.DescribeOptionGroups(params)
+	if err != nil {
+		return fmt.Errorf("Error Describing DB Option Group: %s", err)
+	}
+
+	var option *rds.OptionGroup
+	for _, ogl := range options.OptionGroupsList {
+		if *ogl.OptionGroupName == d.Get("name").(string) {
+			option = ogl
+			break
+		}
+	}
+
+	if option == nil {
+		return fmt.Errorf("Unable to find Option Group: %#v", options.OptionGroupsList)
```

Comment:
Terraform needs the "not found" case to be handled as a non-error so it can make progress in the scenario a resource is deleted out from under it.

There are two ways to accomplish this:

 * The traditional way we've done it is to recognize a not found condition within `Read` and do a log, `d.SetId("")`, and early return.
 * The other way which trades off an extra API call for ease of implementation is to implement [`Exists`](https://github.com/hashicorp/terraform/blob/0cf6ba6f6417cdd5421c7b735950c05bcb4a59d5/helper/schema/resource.go#L71), doing the checking in there. If `Exists` is implemented, then `Read` can proceed assuming the resource exists.

I've been starting to add acceptance tests to verify this behavior. [Here's an example](https://github.com/hashicorp/terraform/blob/f-vault/builtin/providers/vault/resource_policy_test.go#L34-L58).